parts/windows/kuberneteswindowssetup.ps1

```diff
@@ -292,7 +295,9 @@ try
     Get-LogCollectionScripts
     
     Write-KubeClusterConfig -MasterIP $MasterIP -KubeDnsServiceIp $KubeDnsServiceIp
-    
+
+    Install-CredentialProvider -KubeDir $global:KubeDir -CustomCloudContainerRegistryDNSSuffix {{if IsAKSCustomCloud}}"{{ AKSCustomCloudContainerRegistryDNSSuffix }}"{{else}}""{{end}} 

```

Comment:
Prefer to return empty when IsAKSCustomCloud is true in baker.go instead of add this condition check in this ps1 even if it is a template.

Reply:
Considerations here are:
1. AKSCustomCloudContainerRegistryDNSSuffix requires checking CustomCloudEnv empty first, and we have a lot of functions related to CustomCloud, so to keep the change minimal
```
		"AKSCustomCloudContainerRegistryDNSSuffix": func() string {
			return cs.Properties.CustomCloudEnv.ContainerRegistryDNSSuffix
		},
```
2. IsAKSCustomCloud is referenced several times already in window setup script, so it's safe to use